### PR TITLE
Lazyloading margin, reserving ad space, and ad sizing rules.

### PR DESCRIPTION
**Pull request by fabiosantoscode**

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -78,17 +90,23 @@ export default class AnimatedPanel extends React.Component {
...
     if ((window.googletag) && (this.props.adTag)) {
       const googleTag = window.googletag;
       googleTag.cmd.push(() => {
-        const mappingAd = window.googletag.sizeMapping().addSize([ 980, 200 ], [ 1024, 768 ])
-        .addSize([ 0, 0 ], [ 300, 250 ]).build();
-        googleTag.defineSlot(
+        const mappingAd = window.googletag.sizeMapping()
+          .addSize([ 980, 200 ], [ 1024, 768 ])
+          .addSize([ 0, 0 ], [ 300, 250 ])
+          .build();
```
Comment (fabiosantoscode) [inline index.es6]: This maps screen sizes to ad sizes. It says that in tablet size and above, the ad can have at most [ 1024 x 768 ], and below that, the add can have at most [ 300 x 250 ].

File: index.es6 [hunk 2]
```diff
@@ -8,12 +8,17 @@ export default class AnimatedPanel extends React.Component {
...
+      sizes: React.PropTypes.arrayOf(React.PropTypes.array),
+      reserveHeight: React.PropTypes.number,
     };
   }
 
   static get defaultProps() {
     return {
       lazyLoad: true,
+      lazyLoadMargin: 350,
+      sizes: [ [ 60, 60 ], [ 70, 70 ], [ 300, 250 ], [ 1024, 768 ] ]
```
Comment (fabiosantoscode) [inline index.es6]: These are the different ad sizes we accept in this ad spot. For this demo we only use [ 300, 250 ]

**Comment by fabiosantoscode** [comment]
I've responded to that question here https://github.com/the-economist-editorial/component-animatedpanel/pull/8#discussion_r41526658 and here https://github.com/the-economist-editorial/component-animatedpanel/pull/6#discussion_r41490031

**Comment by fabiosantoscode** [comment]
You can look a demo of this PR in fabiosantoscode.github.io/component-animatedpanel

**Comment by angelorepole** [comment]
Looks good to me! Approved

**Pull request merged**